mimic/test/test_core.py

```diff
@@ -92,3 +102,243 @@ def cleanup():
             fake_plugin.dummy_domain_plugin,
             core.domains
         )
+
+
+class CoreApiBuildingTests(SynchronousTestCase):
```

Comment:
This isn't blocking feedback, but it seems like many of the docstrings in this class could be reflowed to 80 columns, instead of taking multiple lines.